drivers/shared/executor/executor_linux_test.go

```diff
@@ -973,3 +975,37 @@ func TestExecutor_SignalCatching(t *testing.T) {
 	must.NoError(t, err)
 	must.Eq(t, specs.StateStopped, status.Status)
 }
+
+// non-default devices must be present in cgroup device rules
+func TestCgroupDeviceRules(t *testing.T) {
+	ci.Parallel(t)
+	testutil.ExecCompatible(t)
+	testExecCmd := testExecutorCommand(t)
+	command := testExecCmd.command
+
+	allocDir := testExecCmd.allocDir
+	defer allocDir.Destroy()
+
+	command.Devices = append(command.Devices,
+		// /dev/fuse is not in the default device list
+		&drivers.DeviceConfig{
+			HostPath:    "/dev/fuse",
+			TaskPath:    "/dev/fuse",
+			Permissions: "rwm",
+		})
+	execInterface := NewExecutorWithIsolation(testlog.HCLogger(t), compute)
+	executor := execInterface.(*LibcontainerExecutor)
+	cfg, err := executor.newLibcontainerConfig(command)
+	must.NoError(t, err)
+	rules := make([]devices.Rule, len(cfg.Cgroups.Devices))
+	for i, rule := range cfg.Cgroups.Devices {
+		rules[i] = *rule
+	}
+	must.True(t, slices.Contains(rules, devices.Rule{
+		Type:        'c',
+		Major:       0xa,
+		Minor:       0xe5,
+		Permissions: "rwm",
+		Allow:       true,
+	}))
```

Comment:
We can trim this down a good bit:

```suggestion
	must.SliceContains(t, cfg.Cgroups.Devices, &devices.Rule{
		Type:        'c',
		Major:       0xa,
		Minor:       0xe5,
		Permissions: "rwm",
		Allow:       true,
	})
```

Reply:
Let me try this and get back to you. When initially writing this I did not get 'must.SliceContains' to behave in a way that made sense.